src/app/shared/directives/duration-validator.directive.ts

```diff
@@ -0,0 +1,8 @@
+import { ValidatorFn, AbstractControl, ValidationErrors } from "@angular/forms";
+
+export function durationValidator(regExp: RegExp): ValidatorFn {
```

Comment:
1. I woulld calll it dateTimePerdiodValidator.
2. Why we need to pass regexp to it. Its hould be emebeded.,
3. No tests :(